compiler/src/main/kotlin/se/ansman/kotshi/generators/DataClassAdapterGenerator.kt

```diff
@@ -158,18 +162,32 @@ class DataClassAdapterGenerator(
             .addParameter(writerParameter)
             .addParameter(value)
 
+        val label = element.getAnnotation(PolymorphicLabel::class.java)?.value
+        val labelKey = types.asElement(element.superclass)?.getAnnotation(Polymorphic::class.java)?.labelKey
```

Comment:
I'm slightly unsure about this part, not sure if there's a more optimal way to achieve this with `javax.lang.` APIs. It's also duplicated between this and `ObjectAdapterGenerator`, but pulling it up into `AdapterGenerator` felt weird

Reply:
I think this is fine. Typically we should only use the kotlinx APIs but for annotations it's fine. What you could do however is to add a check that the superclass is a sealed class since the Polymorphic annotations might be used in more places in the future. Though I don't think it's needed and would slow down processing. 

You also need to traverse all parent classes to read any nested annotations.

Reply:
This seems to be a good place for a helper function since we'll need to do this both for the object adapter and the data class adapter